src/dsp_tools/commands/excel2json/resources.py

```diff
@@ -284,26 +280,53 @@ def _prepare_classes_df(resource_dfs: dict[str, pd.DataFrame]) -> tuple[pd.DataF
             "comment_rm",
         },
     )
-    classes_df = prepare_dataframe(
-        df=classes_df,
-        required_columns=["name"],
-        location_of_sheet="Sheet 'classes' in file 'resources.xlsx'",
-    )
+    resource_dfs = {k: add_optional_columns(v, {"gui_order"}) for k, v in resource_dfs.items()}
     return classes_df, resource_dfs
 
 
-def _validate_excel_file(classes_df: pd.DataFrame, df_dict: dict[str, pd.DataFrame]) -> list[Problem]:
+def _validate_excel_file(classes_df: pd.DataFrame, df_dict: dict[str, pd.DataFrame]) -> ExcelFileProblem | None:
+    problems = _validate_classes_excel_sheet(classes_df)
+    if sheet_problems := _validate_individual_class_sheets(df_dict):
+        problems.extend(sheet_problems)
+    if problems:
+        return ExcelFileProblem("resources.xlsx", problems)
+    return None
+
+
+def _validate_classes_excel_sheet(classes_df: pd.DataFrame) -> list[Problem]:
     if any(classes_df.get(lang) is not None for lang in languages):
         warnings.warn(
             f"The file 'resources.xlsx' uses {languages} as column titles, which is deprecated. "
             f"Please use {[f'label_{lang}' for lang in languages]}"
         )
     problems: list[Problem] = []
-    if missing_super_rows := [int(str(i)) + 2 for i, row in classes_df.iterrows() if not check_notna(row["super"])]:
```

Comment:
This code is not actually gone, it just now uses a previously existing and tested function.